zokrates_core/src/compile.rs

```diff
@@ -162,9 +168,25 @@ impl fmt::Display for CompileErrorInner {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq)]
+pub enum CompileMode {
+    Bin,
```

Comment:
I think we should use something like this here

```rust
#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq)]
pub enum CompileMode {
    #[serde(rename = "bin")]
    Bin,
    #[serde(rename = "lib")]
    Lib
}
```

so it's a bit cleaner in js: `mode: "lib` instead of `mode: "Lib"`. It's also possible to put `#[serde(rename_all = "lowercase")]` before the enum definition. 

We should also update typescript type information file:
https://github.com/Zokrates/ZoKrates/blob/add-compiler-mode-flag/zokrates_js/index.d.ts#L12

```ts
export type CompileMode = "bin" | "lib";

export interface CompileConfig {
   allow_unconstrained_variables?: boolean,
   isolate_branches?: boolean,
   mode?: CompileMode
}
```